JoinColony/colonyCDapp
# Fix: Correctly show users on the permissions page based on team filter

## Description

![all-teams-filter](https://github.com/user-attachments/assets/514a71bb-64af-40fd-b489-464dcd8300d5)

## Testing

1. Add Fiona as a Payer in team General
2. Remove Fiona's permission from team Andromeda
3. Add Amy as a Payer in team Andromeda
4. Add Amy as a Mod in team Serenity
5. Add Fry as an Owner in team General
6. Change Fry's permission in team Andromeda to Arbitration only
7. Set the Permissions page filter to All Teams
8. Verify that:
  - you don't see the "Inherited" badge on any user tile
  - Leela & Fry are both under the Owner category
  - Fiona & Amy are both under the Payer category
9. Set the Permissions page filter to Andromeda
10. Verify that:
  - Leela is under the Owner category and her tile has the "Inherited" badge
  - Fry is under the Custom category and her tile does not have the "Inherited" badge
  - Amy is under the Payer category and her tile does not have the "inherited" badge
  - You don't see Fiona in the results
 11. Set the Permissions page filter to Serenity
 12. Verify that:
   - Both Leela and Fry are under the Owner category and they both have the "Inherited" badge
   - Fiona is under the Payer category and her tile has the "Inherited" badge
   - Amy is under the Mod category and her tile does not have the "inherited" badge

Resolves #2784
